### PR TITLE
LPS-130271 If the form contains any ddm fieldsets await for the data engine validation before the submit

### PR DESCRIPTION
**Pull request by boton**
https://issues.liferay.com/browse/LPS-130271

This PR will be going to forward to the DE team when all is ok.

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
Just started reviewing :)

**Review by AliciaGarciaGarcia — APPROVED** [review]
The only thing I can add it is that if we can re-open the document-type because the field validation is not shown if is failing and it is closed:
image

image

**Comment by boton** [comment]
> The only thing I can add it is that if we can re-open the document-type because the field validation is not shown if is failing and it is closed:

You are right sorry for the delay I want to test it in 7.3.x and work opening the panel in this case. 
I'm going to update the PR and try to open this slider in case of failing validation.

7.3.x
image

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries.jsp [hunk 1]
```diff
@@ -151,113 +151,130 @@ if (portletTitleBasedNavigation) {
...
+								};
 
-										selectedItems.all('input').remove(true);
+								var isDataEngineControlled = Boolean(
+									document.querySelector('[data-ddm-fieldset]')
+								);
 
-										commonFileMetadataContainer.loadingmask.hide();
+								if (isDataEngineControlled) {
+									Liferay.on('ddmFormValid', (event) => {
```
Comment (carloslancha) [inline modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries.jsp:269]: Shouldn't we detach this event at some point?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js [hunk 2]
```diff
@@ -69,7 +69,20 @@ const useFormSubmit = ({apiRef, containerRef}) => {
 							return;
 						}
 
-						Liferay.Util.submitForm(event.target);
+						// TODO: find a better way to communicate to avoid submit than css class
```
Comment (carloslancha) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js:72]: Yup, don't use css classes. Use data attributes instead.

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries.jsp (hunk 1 above)
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries.jsp:269]: Thanks, you are right :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js (hunk 2 above)
Comment (boton) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js:72]: `aui:form` doesn´t support data attributes 😢

Comment (boton) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js]: data used here 0abaaa87ff438216e2f944b615c1f99626b834af

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries.jsp (hunk 1 above)
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/upload_multiple_file_entries.jsp]: Detach events here dea61e9775094b5a00ddd701dfbe60718943d685

**Comment by boton** [comment]
Guys, I update this, @AliciaGarciaGarcia the panel is open on an error in DM, and @carloslancha I use data instead of class and detach the events. If you are right with the changes I'm going to send this to data engine.

**Comment by boton** [comment]
ci:test:sf

**Comment by boton** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
Just started reviewing :)

**Review by carloslancha — APPROVED** [review]
lgtm

**Comment by boton** [comment]
See liferay-data-engine#416.

**Pull request closed without merging**